ext/bigdecimal/bigdecimal.c

```diff
@@ -1854,14 +1854,15 @@ BigDecimal_divide(VALUE self, VALUE r, Real **c, Real **res, Real **div)
 
     BigDecimal_count_precision_and_scale(self, &a_prec, NULL);
     BigDecimal_count_precision_and_scale(rr, &b_prec, NULL);
-    mx = (a_prec > b_prec) ? a_prec : b_prec;
-    mx *= 2;
+    mx = ((a_prec > b_prec) ? a_prec : b_prec) + BIGDECIMAL_DOUBLE_FIGURES;
 
     if (2*BIGDECIMAL_DOUBLE_FIGURES > mx)
         mx = 2*BIGDECIMAL_DOUBLE_FIGURES;
```

Comment:
Precision was `max(a_prec*2, b_prec*2, 32)`.
I think we normally need `max(a_prec, b_prec) + extra` precision, and the `extra` part was `≧16` before.